homeassistant/components/wolflink/__init__.py

```diff
@@ -42,17 +43,40 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     wolf_client = WolfClient(username, password)
 
-    try:
-        parameters = await fetch_parameters(wolf_client, gateway_id, device_id)
-    except InvalidAuth:
-        _LOGGER.debug("Authentication failed")
-        return False
+    parameters = await fetch_parameters_init(wolf_client, gateway_id, device_id)
 
     async def async_update_data():
         """Update all stored entities for Wolf SmartSet."""
         try:
-            values = await wolf_client.fetch_value(gateway_id, device_id, parameters)
-            return {v.value_id: v.value for v in values}
+            nonlocal refetch_parameters
+            nonlocal parameters
+            await wolf_client.update_session()
+            if not wolf_client.fetch_system_state_list(device_id, gateway_id):
+                refetch_parameters = True
+                raise UpdateFailed(
+                    "Could not fetch values from server because device is Offline."
+                )
+            else:
```

Comment:
We can remove this `else:` and outdent below since we raise above.

Reply:
raise appeared after else so you are right else was not necessary.